### PR TITLE
drivers: bluetooth: hci: handle event with high priority flag

### LINKED ISSUE #65892: drivers: bluetooth: hci: h4 high priority packtes ignored if CONFIG_BT_RECV_BLOCKING is not set
Looking at the code below:

https://github.com/zephyrproject-rtos/zephyr/blob/617cb789c34585921c07c883d9739faa62d75ce1/drivers/bluetooth/hci/h4.c#L333-L354

and with actual test, it appears to me that packets with `evt_flags` defined as `BT_HCI_EVT_FLAG_RECV_PRIO` by are ignored/discarded when CONFIG_BT_RECV_BLOCKING is not set.
This typically happens for `BT_HCI_EVT_CMD_COMPLETE` or `BT_HCI_EVT_CMD_STATUS` events making the driver unusable.

### PR DESCRIPTION
**Pull request by msmttchr**
Make sure that events flagged as high priority are handled when `CONFIG_BT_RECV_BLOCKING` is not defined.

Fixes  #65892.

### CODE REVIEW

File: drivers/bluetooth/hci/h4.c [hunk 1]
```diff
@@ -344,9 +344,8 @@ static inline void read_payload(void)
 	    (evt_flags & BT_HCI_EVT_FLAG_RECV_PRIO)) {
 		LOG_DBG("Calling bt_recv_prio(%p)", buf);
 		bt_recv_prio(buf);
-	}
-
-	if (evt_flags & BT_HCI_EVT_FLAG_RECV) {
+	} else if ((evt_flags & BT_HCI_EVT_FLAG_RECV) ||
+		   (evt_flags & BT_HCI_EVT_FLAG_RECV_PRIO)) {
```
Comment (jhedberg) [inline drivers/bluetooth/hci/h4.c:348]: That's a good catch. However, the only "receive" flags an HCI packet can have are either "normal" or "priority", so I don't think you need to explicitly test for these. Instead you can just extend the first if-statement with basic `} else {` branch.

**Review by jhedberg — CHANGES_REQUESTED** [review]
Looks like a good fix, but I think you can do it slightly simpler. See my inline comment.

File: drivers/bluetooth/hci/h4.c (hunk 1 above)
Comment (msmttchr) [reply drivers/bluetooth/hci/h4.c:348]: Indeed, this was my first choice, but I have checked the function  bt_hci_evt_get_flags and it seems that there are cases when both flags are set.

Comment (jhedberg) [reply drivers/bluetooth/hci/h4.c:348]: Both flags being set means that the host will process the event in both priority and normal context. It doesn't really make any difference here though - my point was that it doesn't matter what flags (or how many) are set, you still want to just call `bt_recv()` in the new else-branch.

Comment (jhedberg) [reply drivers/bluetooth/hci/h4.c:348]: Correction, call `net_buf_put()` in the else branch (the actual `bt_recv()` happens from another thread)

Comment (msmttchr) [reply drivers/bluetooth/hci/h4.c:348]: Thanks for clarification, I will make the changes.
I still have a doubt, with my proposal or with your simplified version, there is a change in behavior in the following condition
- `CONFIG_BT_RECV_BLOCKING` set
- `BT_HCI_EVT_FLAG_RECV_PRIO` and `BT_HCI_EVT_FLAG_RECV` both set

Current version is calling both `bt_recv_prio` and `bt_recv` (via `net_buf_put`), while new version is only calling `bt_recv_prio`. Is this correct ?

**Review by jhedberg — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**